File: Kotlin/LeetCode0014.kt
class LeetCode0014 {
    fun longestCommonPrefix(strs: Array<String>): String {
        if(strs.isEmpty()) { return ""}
        strs.sort()
        var shortest = strs[0]
        for(str in strs) {
             while(str.indexOf(shortest) != 0) {
                 shortest = shortest.substring(0,shortest.length-1);
             }
            
        }
        return shortest
    }
}

File: Go/Leetcode0014.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var input = []string{"flower", "flow", "flight"}
	fmt.Print(longestCommonPrefix(input))
}

func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	sortByLength := func(i, j int) bool {
		return len(strs[i]) < len(strs[j])
	}
	sort.Slice(strs, sortByLength)
	//shortest :=
	// for _, str := range strs {

	// }
	return "x"
}
